internal/storage/service_test.go

```diff
@@ -2033,3 +2033,126 @@ func TestUpdateTask(t *testing.T) {
 		assert.DeepEqual(t, re, task)
 	})
 }
+
+func TestCreateDeletionRequest(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Creates a DeletionRequest", func(t *testing.T) {
+		t.Parallel()
+
+		attrs := &setUpAttrs{}
+		svc := setUpService(t, attrs)
+		ctx := context.Background()
+
+		dr := &types.DeletionRequest{
+			UUID:        uuid.New(),
+			AIPUUID:     uuid.New(),
+			Reason:      "Reason",
+			Status:      enums.DeletionRequestStatusPending,
+			RequestedAt: time.Now(),
+		}
+		perErr := errors.New("persistence error")
+
+		attrs.persistenceMock.
+			EXPECT().
+			CreateDeletionRequest(ctx, dr).
+			Return(perErr)
+
+		err := svc.CreateDeletionRequest(ctx, dr)
+		assert.ErrorIs(t, err, perErr)
+	})
+}
+
+func TestUpdateDeletionRequest(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Updates a DeletionRequest", func(t *testing.T) {
+		t.Parallel()
+
+		attrs := &setUpAttrs{}
+		svc := setUpService(t, attrs)
+		ctx := context.Background()
+
+		drID := 1
+		dr := &types.DeletionRequest{
+			DBID:        drID,
+			UUID:        uuid.New(),
+			AIPUUID:     uuid.New(),
+			Reason:      "Updated reason",
+			Status:      enums.DeletionRequestStatusApproved,
+			RequestedAt: time.Now(),
+			ReviewedAt:  time.Now(),
+			Reviewer:    "reviewer@example.com",
+			ReviewerISS: "issuer",
+			ReviewerSub: "subject",
+		}
+		updater := func(dr *types.DeletionRequest) (*types.DeletionRequest, error) { return dr, nil }
```

Comment:
I'm concerned about defining an ad-hoc updater for testing the deletion request, rather than testing a standard implementation.  I've found errors when testing updaters in the past due missing fields or unexpected empty or nil values.

Reply:
See my comments in https://github.com/artefactual-sdps/enduro/pull/1195#discussion_r2067946835. In here, we only pass the updater to the persistence client without doing anything with it or worrying about its output (for that we test the actual definition in the [activity tests](https://github.com/artefactual-sdps/enduro/pull/1199/files#diff-00d6d8ba6d938f78d56ea699fce7ae563cf581cb33d2a07a6c98afba089d353dR291-R325)).